src/content/guide/how-to-build-a-product/device-groups.md

```diff
@@ -0,0 +1,209 @@
+in your fleet to a group. Note that **a device can belong to multiple
+groups**, as you will likely need to segment your fleet
+along many different axes.
+
+To assign a single device to a group, click on it from your product's
+devices view (<i class="im-devices-icon"></i>). Then, click the **+ Add
+groups** button as shown in the device header:
+
+<img class="full-width" alt="Tag device to group" src="/assets/images/device-groups/tag-device-to-group.png"/>
+
+The page will enter into an edit mode. Apply groups to the device
+using the dropdown, and click the **Done** button to save. You can tag
+devices with existing groups, or create a new group and apply it to the
+device. You can come back here and click the **Edit** button on a device
+to update its group memberships at any time.
+
+You can also assign many devices to a group at the same time. If you
+have grouping enabled for your product, you should see checkboxes next
+to each approved device visible on your product's device view. Select
+some devices, and click the **Edit groups** button that appears:
+
+<img class="full-width" alt="Batch edit groups"
+src="/assets/images/device-groups/batch-edit-groups.png"/>
+<p class="caption">Edit groups of many devices at once
+using the Console</p>
+
+You will then be presented with a modal that will allow you to batch
+update groups of the selected devices. **Note that editing groups for a
+device may result in the device being targeted for an OTA firmware
+update**, due to product firmware releases by group.
+
+_Quick tip_: If you prefer, you can also [call the Particle API](!insert
+group link) directly to tag a device to a group.
+
+_Quick tip_: After tagging devices with groups, you can use the Console
+or the [Particle API](!insert link) to filter product devices by group.
+
+## Firmware Release by Group
+
+With devices segmented into groups, it is now possible to gain more
+control and flexibility over the firmware release process. If you are
+unfamiliar with how to rollout firmware to a product fleet, please check
+out the [Particle Console
+Guide](/guide/tools-and-features/console/#rollout-firmware) before
+continuing.
+
+With grouping enabled, you can *release a product firmware to one or
+more groups*,
+or choose to mark the version as the *Product default* release.
+Releasing firmware to a group will target a only subset of the product
+fleet to receive the binary (those devices that belong to that group).
+
+Marking as the Product default, in
+contrast, sets the firmware as the default firmware version
+available to all devices in the fleet to download and run.
+The specific firmware chosen to be delivered to a
+given device is determined by *precedence rules*, which you can read about in
+the next section.
+
+### Firmware Precedence Rules
+
+Devices in your fleet will be targeted to
+receive a version of product firmware according to these precedence
+rules:
+
+- A device that has been **individually locked** to a version of product
```

Comment:
One more rule with higher precedence: a development device never receives automatic updates of product firmware.